# Data Science Salaries 2023

## Data Gathering

In [7]:
!mkdir ~/.kaggle
!touch ~/.kaggle/kaggle.json

api_token = {"username":"firtolatte","key":"7a0840647e54920c8ab6a6acc1894646"}

import json

with open('/root/.kaggle/kaggle.json', 'w') as file:
    json.dump(api_token, file)

!chmod 600 ~/.kaggle/kaggle.json
!kaggle datasets download -d arnabchaki/data-science-salaries-2023 --force

mkdir: cannot create directory ‘/root/.kaggle’: File exists
  0%|                                               | 0.00/25.4k [00:00<?, ?B/s]
100%|███████████████████████████████████████| 25.4k/25.4k [00:00<00:00, 771kB/s]


In [8]:
# importing the zipfile module 
from zipfile import ZipFile 
  
# loading the temp.zip and creating a zip object 
with ZipFile("data-science-salaries-2023.zip", 'r') as zObject: 
  
    # Extracting all the members of the zip  
    # into a specific location. 
    zObject.extractall( 
        path="data/") 

## Importing Libraries

### Libraries to install:
- kaggle
- country_converter
- matplotlib
- seaborn
- plotly
- wordcloud
- nltk
- scikit-learn
- xgboost

In [21]:
import pandas as pd
import numpy as np
import country_converter as coco
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.figure_factory as ff
import plotly.graph_objects as go
from wordcloud import WordCloud
import warnings
warnings.filterwarnings('ignore')
import nltk

%matplotlib inline

In [22]:
df = pd.read_csv('data/ds_salaries.csv')
df.drop(df[['salary','salary_currency']], axis = 1, inplace = True)

In [23]:
print(df.shape)
df.head()

(3755, 9)


work_year experience_level employment_type                 job_title  \
0       2023               SE              FT  Principal Data Scientist   
1       2023               MI              CT               ML Engineer   
2       2023               MI              CT               ML Engineer   
3       2023               SE              FT            Data Scientist   
4       2023               SE              FT            Data Scientist   

   salary_in_usd employee_residence  remote_ratio company_location  \
0          85847                 ES           100               ES   
1          30000                 US           100               US   
2          25500                 US           100               US   
3         175000                 CA           100               CA   
4         120000                 CA           100               CA   

  company_size  
0            L  
1            S  
2            S  
3            M  
4            M

So, we have 9 columns with 3755 rows:

3 numeric columns : work_year, salary_in_usd,remote_ratio.

6 categorical columns : experience_level,employment_type, job_title, employee_residense, company_location, company_size.

In [25]:
df.isnull().sum()

work_year             0
experience_level      0
employment_type       0
job_title             0
salary_in_usd         0
employee_residence    0
remote_ratio          0
company_location      0
company_size          0
dtype: int64

### Experience Level
There's 4 categorical values in column 'Experience Level', each are:

EN, which refers to Entry-level / Junior.

MI, which refers to Mid-level / Intermediate.

SE, which refers to Senior-level / Expert.

EX, which refers to Executive-level / Director.

In [27]:
df['experience_level'] = df['experience_level'].replace('EN','Entry-level/Junior')
df['experience_level'] = df['experience_level'].replace('MI','Mid-level/Intermediate')
df['experience_level'] = df['experience_level'].replace('SE','Senior-level/Expert')
df['experience_level'] = df['experience_level'].replace('EX','Executive-level/Director')

ex_level = df['experience_level'].value_counts()
fig = px.treemap(ex_level, path = [ex_level.index], values = ex_level.values, 
                title = 'Experience Level')
fig.show()